src/UI/Components/Diagram/zoomHandler/zoomHandler.ts

```diff
@@ -0,0 +1,301 @@
+import { ui } from "@inmanta/rappid";
+import { words } from "@/UI/words";
+import exitFullscreen from "../icons/exit-fullscreen.svg";
+import fitToScreen from "../icons/fit-to-screen.svg";
+import requestFullscreen from "../icons/request-fullscreen.svg";
+
+/**
+ * Interface for a button with an icon and a tooltip.
+ *
+ * This interface extends the ui.widgets.button interface and adds methods for setting the icon and the tooltip of the button.
+ */
+interface IconButton extends ui.widgets.button {
+  setIcon(icon: string): void;
+  setTooltip(tooltip: string): void;
+}
+
+/**
+ * IconButton class
+ *
+ * This class extends the ui.widgets.button class and represents a button with an icon.
+ * It provides methods for setting the icon and the tooltip of the button.
+ *
+ */
+const IconButton = ui.widgets.button.extend({
+  render: function () {
+    const size = this.options.size || 20;
+    const imageEl = document.createElement("img");
+
+    imageEl.style.width = `${size}px`;
+    imageEl.style.height = `${size}px`;
+    this.el.appendChild(imageEl);
+    this.setIcon(this.options.icon);
+    this.setTooltip(this.options.tooltip);
+
+    return this;
+  },
+
+  /**
+   * Sets the icon of the element.
+   *
+   * This method is responsible for setting the source of the image element within the current element to the provided icon.
+   *
+   * @param {string} icon - The source of the icon to set. Defaults to an empty string, which will clear the current icon.
+   */
+  setIcon: function (icon = "") {
+    this.el.querySelector("img").src = icon;
+  },
+
+  /**
+   * Sets the tooltip of the element.
+   *
+   * This method is responsible for setting the tooltip of the current element to the provided tooltip.
+   * It also sets the position of the tooltip.
+   *
+   * @param {string} tooltip - The text of the tooltip to set. Defaults to an empty string, which will clear the current tooltip.
+   * @param {string} direction - The position of the tooltip. Defaults to "right".
+   */
+  setTooltip: function (tooltip = "", direction = "right") {
+    this.el.dataset.tooltip = tooltip;
+    this.el.dataset.tooltipPosition = direction;
+  },
+});
+
+/**
+ * ZoomHandlerService class
+ *
+ * This class is responsible for managing the navigation of the canvas.
```

Comment:
Paning would be a better word than navigation :) Navigation implies routing, or boats, or GPS